python/graphstorm/gconstruct/config_conversion/gsprocessing_converter.py

```diff
@@ -0,0 +1,329 @@
+                        ],
+                    }
+                elif gsp_transformation_dict["name"] == "categorical":
+                    gconstruct_transformation_dict["name"] = "to_categorical"
+                elif gsp_transformation_dict["name"] == "multi-categorical":
+                    gconstruct_transformation_dict["name"] = "to_categorical"
+                    gconstruct_transformation_dict["separator"] = (
+                        gsp_transformation_dict["kwargs"]["separator"]
+                    )
+                elif gsp_transformation_dict["name"] == "huggingface":
+                    if gsp_transformation_dict["kwargs"]["action"] == "embedding_hf":
+                        action_name = "bert_hf"
+                    elif gsp_transformation_dict["kwargs"]["action"] == "tokenize_hf":
+                        action_name = "tokenize_hf"
+                    else:
+                        raise ValueError("Unexpected huggingface action {}"
+                                         .format(gsp_transformation_dict["kwargs"]["action"]))
+                    gconstruct_transformation_dict = {
+                        "name": action_name,
+                        "bert_model": gsp_transformation_dict["kwargs"]["hf_model"],
+                        "max_seq_length": gsp_transformation_dict["kwargs"][
+                            "max_seq_length"
+                        ],
+                    }
+                elif gsp_transformation_dict["name"] == "edge_dst_hard_negative":
+                    gconstruct_transformation_dict["name"] = "edge_dst_hard_negative"
+                    if "separator" in gsp_transformation_dict["kwargs"]:
+                        gconstruct_transformation_dict["separator"] = (
+                            gsp_transformation_dict["kwargs"]["separator"]
+                        )
+                else:
+                    raise ValueError(
+                        "Unsupported GSProcessing transformation name: "
+                        f"{gsp_transformation_dict['name']}"
+                    )
+                ### END TRANSFORMATION ALTERNATIVES ###
+            else:
+                gconstruct_transformation_dict["name"] = "no-op"
+
+            gconstruct_feat_dict["transform"] = gconstruct_transformation_dict
+            gconstruct_feats_list.append(gconstruct_feat_dict)
+
+        return gconstruct_feats_list
+
+    @staticmethod
+    def convert_nodes(nodes_entries) -> list[NodeConfig]:
+        res = []
+        for n in nodes_entries:
+            # type, column id
+            node_type, node_col = n["type"], n["column"]
+            # format
+            # Assume the format is already valid in GSProcessing
+            node_format = n["data"]["format"]
+            if "separator" not in n["data"]:
+                node_separator = None
+            else:
+                node_separator = n["data"]["separator"]
+
+            # files
+            # Do we support S3 in GConstruct?
```

Comment:
We do not, the assumption here should be that all paths are relative. 

We need practical examples of how the converter would be used locally and on SageMaker.